lib/MusicBrainz/Server/Entity/Medium.pm

```diff
@@ -182,7 +184,7 @@ around TO_JSON => sub {
         name        => $self->name,
         position    => $self->position,
         release_id  => $self->release_id,
-        track_count => 0 + $self->track_count,
+        track_count => defined $track_count ? (0 + $track_count) : undef,
```

Comment:
I think using `undef` makes sense because some mediums actually do have a track count of 0.  This lets us distinguish real cases from cases where it's simply not loaded.

The implications are that we need to handle `null` in some places, but I didn't find any uses in files that aren't Flow-typed, so they should be all handled in that patch.

Reply:
Thaanks, it makes sense. It would be still nice to write all of this in commit message.

Reply:
Forgot to push the commit message fixes earlier, sorry. :)

Reply:
No problem, thanks :)